libraries/core_libs/consensus/include/dag/dag.hpp

```diff
@@ -63,8 +63,15 @@ class Dag {
   using edge_index_map_t = boost::property_map<graph_t, boost::edge_index_t>::type;
 
   friend DagManager;
+
   explicit Dag(blk_hash_t const &genesis, addr_t node_addr);
   virtual ~Dag() = default;
+
+  Dag(const Dag &) = default;
+  Dag(Dag &&) = default;
+  Dag &operator=(const Dag &) = default;
+  Dag &operator=(Dag &&) = default;
```

Comment:
why is this needed? shouldn't be constructed by default, without this definitions?